tests/topotests/lib/evpn.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: ISC
+#
+# Copyright (c) 2025 by 6WIND
+# Used Copyright (c) 2018 by Network Device Education Foundation, Inc.
+# ("NetDEF") in this file.
+#
+
+from functools import partial
+from lib import topotest
+from lib.bgp import verify_bgp_rib
+
+from lib.topolog import logger
+
+
+def evpn_ping_router(pingrouter, ipv4_only=False, ipv6_only=False):
+    """
+    internal function to check ping between r1 and r2
+    """
+    # Check IPv4 and IPv6 connectivity between r1 and r2 ( routing vxlan evpn)
+    if not ipv6_only:
+        logger.info("Check Ping IPv4 from R1(vrf-101) to R2(vrf-101, 10.0.101.2)")
+        output = pingrouter.run("ip netns exec vrf-101 ping 10.0.101.2 -f -c 1000")
+        logger.info(output)
+        if "1000 packets transmitted, 1000 received" not in output:
+            assertmsg = "expected ping IPv4 from R1(vrf-101) to R2(vrf-101, 10.0.101.2) should be ok"
+            assert 0, assertmsg
+        else:
+            logger.info(
+                "Check Ping IPv4 from R1(vrf-101) to R2(vrf-101, 10.0.101.2) OK"
+            )
+
+    if not ipv4_only:
+        logger.info("Check Ping IPv6 from  R1(vrf-101) to R2(vrf-101, fd01::2)")
+        output = pingrouter.run("ip netns exec vrf-101 ping fd01::2 -f -c 1000")
+        logger.info(output)
+        if "1000 packets transmitted, 1000 received" not in output:
+            assert (
+                0
+            ), "expected ping IPv6 from R1(vrf-101) to R2(vrf-101, fd01::2) should be ok"
+        else:
+            logger.info("Check Ping IPv6 from R1(vrf-101) to R2(vrf-101, fd01::2) OK")
+
+
+def evpn_print_nexthop_rmac(tgen, router):
+    output = tgen.gears[router].vtysh_cmd("show evpn next-hops vni all", isjson=False)
+    logger.info("==== result from {} show evpn next-hops vni all".format(router))
+    logger.info(output)
+    output = tgen.gears[router].vtysh_cmd("show evpn rmac vni all", isjson=False)
+    logger.info("==== result from {}: show evpn rmac vni all".format(router))
+    logger.info(output)
+
+
+def evpn_check_nexthop(
+    router,
+    vni=101,
+    ipv4="192.168.0.1",
+    ipv6="::ffff:192.168.0.1",
+    prefix_ipv4="10.0.101.1/32",
+    prefix_ipv6="fd01::1/128",
+    expected_paths=1,
+):
```

Comment:
same idea as above: does it make sense to have the defaults (test specific IPs in a lib) here? probably not defaulting and make the call from the tests use the explicit values makes it usable (and obvious) in more cases

Reply:
The initial idea was to incrementally change the API.
This said, it is in lib/ and it has global impact (an alternative would have been to use the same test and move the library inside), so I removed the default settings, except for excepted_path.